Realm/ObjectServerTests/RealmServer.swift

```diff
@@ -519,15 +519,16 @@ public class RealmServer: NSObject {
         let binDir = Self.buildDir.appendingPathComponent("bin").path
         let libDir = Self.buildDir.appendingPathComponent("lib").path
         let binPath = "$PATH:\(binDir)"
+        let env = [
+            "PATH": binPath,
+            "DYLD_LIBRARY_PATH": libDir
```

Comment:
DYLD_LIBRARY_PATH was always the correct thing here but LD_LIBRARY_PATH worked until macOS 12.